src/validation-dispatcher/single-field-dispatcher.ts

```diff
@@ -2,11 +2,20 @@ import {
   FieldValidation,
   ValidationResult,
   createDefaultValidationResult,
+  FieldValidationFunctionAsync,
 } from '../model';
 import { arrayContainsEntries, isFunction, isUndefinedOrNull } from '../helper';
+import { convertFieldValidationToAsyncIfNeeded } from '../mapper';
 
-const getValidationFn = (fieldValidation: FieldValidation) =>
-  isFunction(fieldValidation) ? fieldValidation : fieldValidation.validator;
+const getValidationFn = (
+  fieldValidation: FieldValidation
+): FieldValidationFunctionAsync => {
+  const validationFn = isFunction(fieldValidation)
+    ? fieldValidation
+    : fieldValidation.validator;
+
+  return convertFieldValidationToAsyncIfNeeded(validationFn);
```

Comment:
Why convert it again? I think is not necessary

Reply:
Unit test failed, have to take deeper looko

Reply:
Same as prior comment we could create this FieldValidationStrict what do you think?

Reply:
Great